src/core/core.scale.js

```diff
@@ -468,28 +462,28 @@ var Scale = Element.extend({
 		var ticks = me.getTicks();
 		var opts = me.options;
 		var tickOpts = opts.ticks;
-		var scaleLabelOpts = opts.scaleLabel;
 		var gridLineOpts = opts.gridLines;
-		var display = me._isVisible();
 		var position = opts.position;
 		var isHorizontal = me.isHorizontal();
+		var thickness;
 
-		// Width
-		if (isHorizontal) {
-			minSize.width = me.maxWidth;
-		} else if (display) {
-			minSize.width = getTickMarkLength(gridLineOpts) + getScaleLabelHeight(scaleLabelOpts);
+		if (!me._isVisible()) {
+			me.width = 0;
+			me.height = 0;
+			return;
```

Comment:
This would make it much cleaner:
```js
setDimensions: function() {
	// ...

	me.minSize = {
		width: 0,
		height: 0
	};
},

// ...

fit: function() {
	// ...

	var minWidth, minHeight;

	// ...

	if (!me._isVisible()) {
		me.width = me.height = 0;
		return;
	}

	// ...

	if (isHorizontal) {
		minWidth = me.maxWidth;
		minHeight = thickness;
	} else {
		minWidth = thickness;
		minHeight = me.maxHeight;
	}

	// ...

			minHeight = Math.min(me.maxHeight, minHeight + labelHeight + tickPadding);

	// ...

			minWidth = Math.min(me.maxWidth, minWidth + labelWidth);

	// ..

	me.width = minSize.width = minWidth;
	me.height = minSize.height = minHeight;
},
```

Reply:
I made most of the change. I changed the way `minSize` is set to be slightly different than suggested. I wanted to keep it all being done in this method because I think having to look across two methods to understand how the value is set makes it harder to follow

Reply:
The reason why I moved `minSize` initialization to `setDimensions` is to make it consistent with other layout box types (= title and legend). Do you still prefer to keep it in `fit`?

Reply:
I've added those files to this PR and made the same changes there to keep them consistent. Does that work for you?

Reply:
Labels are built and events fired between `setDimensions` and `fit`. With this new change `minSize` is undefined in between (title/legend). That could be considered breaking change. I don't have any strong feelings about that though 😄

Reply:
Yeah, that's technically true. They never held actual values, so I imagined no one would rely on them and that the reduced code and improved readability would be more important